src/main/java/net/fabricmc/loom/configuration/providers/forge/MinecraftPatchedProvider.java

```diff
@@ -633,13 +549,35 @@ public void applyLoomPatchVersion(Path target) throws IOException {
 		}
 	}
 
-	@Override
-	public Path getMergedJar() {
-		return minecraftMergedPatchedJar.toPath();
+	public Path getMinecraftPatchedSrgJar() {
+		return minecraftPatchedSrgJar;
+	}
+
+	public Path getMinecraftPatchedJar() {
+		return minecraftPatchedJar;
+	}
+
+	public enum Type {
+		CLIENT_ONLY("client", "client", (patch, userdev) -> patch.clientPatches),
+		SERVER_ONLY("server", "server", (patch, userdev) -> patch.serverPatches),
+		MERGED("merged", "joined", (patch, userdev) -> userdev.joinedPatches);
+
+		private final String id;
+		private final String mcpId;
+		private final BiFunction<PatchProvider, ForgeUserdevProvider, Path> patches;
+
+		Type(String id, String mcpId, BiFunction<PatchProvider, ForgeUserdevProvider, Path> patches) {
+			this.id = id;
+			this.mcpId = mcpId;
+			this.patches = patches;
+		}
 	}
 
-	@Override
-	public List<Path> getMinecraftJars() {
-		return List.of(minecraftMergedPatchedJar.toPath());
+	public interface MinecraftProviderBridge {
```

Comment:
this is quite awful, but can we move it out this class at least, and change to Path since we are calling toPath everytime we use these methods anyways

Reply:
the whole interface is :crab: now